File: app/src/main/java/me/devsaki/hentoid/json/core/JsonSiteSettings.java
package me.devsaki.hentoid.json.core;

import java.util.List;
import java.util.Map;

public class JsonSiteSettings {

    public Map<String, JsonSite> sites;

    public static class JsonSite {
        public Boolean useMobileAgent = null;
        public Boolean useHentoidAgent = null;
        public Boolean useWebviewAgent = null;
        public Boolean hasBackupURLs = null;
        public Boolean hasCoverBasedPageUpdates = null;
        public Boolean useCloudflare = null;
        public Integer parallelDownloadCap = null;
        public Integer requestsCapPerSecond = null;
        public Integer bookCardDepth = null;
        public List<String> bookCardExcludedParentClasses = null;
    }
}

File: app/src/main/java/me/devsaki/hentoid/json/core/JsonSiteSettings.kt
package me.devsaki.hentoid.json.core

data class JsonSiteSettings(
    val sites: Map<String, JsonSite>
) {
    data class JsonSite(
        val useMobileAgent: Boolean?,
        val useHentoidAgent: Boolean?,
        val useWebviewAgent: Boolean?,
        val hasBackupURLs: Boolean?,
        val hasCoverBasedPageUpdates: Boolean?,
        val useCloudflare: Boolean?,
        val hasUniqueBookId: Boolean?,
        val parallelDownloadCap: Int?,
        val requestsCapPerSecond: Int?,
        val bookCardDepth: Int?,
        val bookCardExcludedParentClasses: List<String>?,
        val galleryHeight: Int?
    )
}
